src/flow/src/expr/func.rs

```diff
@@ -0,0 +1,104 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common_time::DateTime;
+use datatypes::data_type::ConcreteDataType;
+use datatypes::types::cast;
+use datatypes::types::cast::CastOption;
+use datatypes::value::Value;
+use hydroflow::bincode::Error;
+use serde::{Deserialize, Serialize};
+use snafu::ResultExt;
+
+use super::ScalarExpr;
+use crate::expr::error::CastValueSnafu;
+use crate::expr::InvalidArgumentSnafu;
+// TODO(discord9): more function & eval
+use crate::{
+    expr::error::{EvalError, TryFromValueSnafu, TypeMismatchSnafu},
+    repr::Row,
+};
+
+/// UnmaterializableFunc is a function that can't be eval independently,
+/// and require special handling
+#[derive(Ord, PartialOrd, Clone, Debug, Eq, PartialEq, Serialize, Deserialize, Hash)]
+pub enum UnmaterializableFunc {
+    Now,
+    CurrentDatabase,
+    CurrentSchema,
```

Comment:
Are `CurrentSchema` and `CurrentDatabase` not the same thing?

Reply:
`CurrentDatabase` removed

> Are `CurrentSchema` and `CurrentDatabase` not the same thing?